### DISCUSSION TITLE
How to use rector with formatted code defined by PHP Code Sniffer 3 configuration file

### DISCUSSION THREAD

**Discussion by interduo** [rectorphp/rector#8421] [Q&A] [answered]
Is it possible to make changes by rector using some coding standard eg PSR2 supplied by PHP Code Sniffer 3 configuration file?

**Comment by TomasVotruba** [comment, marked as answer]
Hi, the AST printer is independent on coding standard. For custom coding standard, it's better to use your favorite CS tool.

We use [ECS](https://github.com/easy-coding-standard/easy-coding-standard) as easy to setup and use:

```php
# ecs.php
use Symplify\EasyCodingStandard\Config\ECSConfig;

return ECSConfig::configure()
    ->withPaths([
        __DIR__ . '/src',
    ])
    ->withPreparedSets(psr12: true);

```
